### PR TITLE
Setting up Freshtab experiment

### PR DESCRIPTION
**Pull request by chrmod**
Under `freshtab` branch we are going to experiment with react-native to figure out:
* is it suitable platform for fast prototyping of new tab features
* what is it performance/startup cost

This PR, sets up minimalistic configuration for ad-hoc builds only.

IMPORTANT: Those changes should not land on `master` until investigations are finished.

### CODE REVIEW

File: mozilla-release/mobile/android/app/build.gradle [hunk 1]
```diff
@@ -282,6 +289,7 @@ dependencies {
     implementation "com.android.support:design:$support_library_version"
     implementation "com.android.support:customtabs:$support_library_version"
     implementation "com.android.support:palette-v7:$support_library_version"
+    implementation "com.facebook.react:react-native:0.56.1"
```
Comment (ravjit-cliqz) [inline mozilla-release/mobile/android/app/build.gradle]: cliqz start ... cliqz end

File: mozilla-release/mobile/android/app/src/main/res/xml/preferences_advanced.xml [hunk 2]
```diff
@@ -69,11 +69,15 @@
...
             android:title="@string/pref_splash_screen_timeout" />
+        <SwitchPreference android:key="pref.freshtab.enabled"
+                          android:title="Enable Freshtab" />
         <!-- Cliqz end -->
 
         <SwitchPreference android:key="devtools.remote.usb.enabled"
                           android:title="@string/pref_developer_remotedebugging_usb" />
 
+
+
```
Comment (ravjit-cliqz) [inline mozilla-release/mobile/android/app/src/main/res/xml/preferences_advanced.xml]: blank lines

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java [hunk 3]
```diff
@@ -685,6 +697,13 @@ public boolean onKeyUp(int keyCode, KeyEvent event) {
             ThreadUtils.getUiHandler().removeCallbacks(mCheckLongPress);
         }
 
+        /* Cliqz start */
+        if (keyCode == KeyEvent.KEYCODE_MENU && mReactInstanceManager != null) {
+            mReactInstanceManager.showDevOptionsDialog();
+            return true;
+        }
+        /* Cliqz end */
```
Comment (ravjit-cliqz) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java]: We dont need this. Have it in the menu already

File: Jenkinsfile.ad-hoc [hunk 4]
```diff
@@ -92,7 +93,7 @@ def baseImage
 def codeImage
 def id
 node ('master'){
-    def dockerTag = "" 
+    def dockerTag = ""
     stage('Checkout') {
         checkout scm
```
Comment (sharath-cliqz) [inline Jenkinsfile.ad-hoc]: Won't this always checkout 'master' and ignore the parameter ?

**Pull request merged**